service/matching/forwarder.go

```diff
@@ -141,30 +145,43 @@ func (fwdr *Forwarder) ForwardTask(ctx context.Context, task *internalTask) erro
 
 	var err error
 
+	// todo: Vet recomputing ScheduleToStart and rechecking expiry here
+	expiryGo, err := types.TimestampFromProto(task.event.TaskData.Expiry)
```

Comment:
Will switch, I started the timestamp type halfway through  this

Reply:
Actually, this is because I want to validate here and the Timestamp is currently not validated. This line here will go away after History stamps the time